### ISSUE TITLE
Desktop apps fail to load resources in some cases

### ISSUE THREAD

**Issue by whitphx** [whitphx/stlite#413]
## The loading animation GIF in sub pages
When the app is re-run in sub pages, the following error occurs at loading the GIF.
```
GET file:///path/to/stlite/packages/desktop/build/index.html/static/media/icon_running.de1119da.gif net::ERR_FILE_NOT_FOUND
```

## Lazy-imported components in sub pages
When a lazily loaded component is used in a sub page for the first time, loading the chunked JS file fails.

```python
# pages/subpage.py
import streamlit as st
import pandas as pd
import numpy as np

@st.experimental_memo
def load_data():
    df = pd.DataFrame(np.random.randn(50, 3), columns=["a", "b", "c"])
    return df

chart_data = load_data()

st.bar_chart(chart_data)
```

```python
# The top page does not use the same component
```

Then the error would be
```
bootstrap:195          GET file:///path/to/stlite/packages/desktop/build/index.html/static/js/2.f12b74fe.chunk.js net::ERR_FILE_NOT_FOUND
```

---

### This bug is not reproduced in the dev mode
Because the frontend resources are served from a web server (`http://localhost:3000`) in the dev mode and such path resolutions work well same as the web versions.

**Comment by whitphx** [comment]
These problems can be solved by the followings:
* Add `<base href="/" />` to `index.html`
* Apply [this change](https://github.com/electron/electron/issues/4612#issuecomment-189116655 ) to `electron/main.ts`

However, it results in another problem of loading custom component resources.

**Comment by whitphx** [comment]
The reason may be the `host` become `""` on the Electron runtime while it would be something like `localhost:8000` when served as a web page.

**Comment by whitphx** [comment]
It was expected to be solved with v0.17.0, but it seems to still exist.
